### PR TITLE
[RDBMS] Fix operations.py file installing dependencies in CloudShell

### LINKED ISSUE #21535: Unable to add az extension rdbms-connect in cloud shell
## Describe the bug
I am unable to add az extension rdbms-connect from the  cloud shell.

**Command Name**
`az extension add`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
[Errno 2] No such file or directory: 'sudo'
Traceback (most recent call last):
  File "/opt/az/lib/python3.8/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 658, in execute
    raise ex
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 692, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.8/site-packages/azure/cli/command_modules/extension/custom.py", line 16, in add_extension_cmd
    return add_extension(cli_ctx=cmd.cli_ctx, source=source, extension_name=extension_name, index_url=index_url,
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/extension/operations.py", line 336, in add_extension
    extension_name = _add_whl_ext(cli_ctx=cmd_cli_ctx, source=source, ext_sha256=ext_sha256,
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/extension/operations.py", line 115, in _add_whl_ext
    _install_deps_for_psycopg2()
  File "/opt/az/lib/python3.8/site-packages/azure/cli/core/extension/operations.py", line 224, in _install_deps_for_psycopg2
    exit_code = subprocess.call(apt_update_cmd, True)
  File "/opt/az/lib/python3.8/subprocess.py", line 340, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/az/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/az/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'sudo'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az extension add -n rdbms-connect`

## Expected Behavior
Extension install without failures.

## Environment Summary
```
Linux-5.4.0-1070-azure-x86_64-with-glibc2.28 (Cloud Shell), Common Base Linux Delridge (quinault)
Python 3.8.12
Installer: DEB

azure-cli 2.34.0 *

Extensions:
ai-examples 0.2.5
ssh 1.0.0

Dependencies:
msal 1.16.0
azure-mgmt-resource 20.0.0
```
## Additional Context
I ran into this problem while attempting to automate the DB deployment for a azure postgres flexible server using the command:
`az postgres flexible-server execute <args>`

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

Fix #21535

Rework https://github.com/Azure/azure-cli/pull/21294: Skip CLI behavior trying to install dependencies with 'sudo' commands in the Cloud Shell

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 1]
```diff
@@ -213,18 +213,20 @@ def _install_deps_for_psycopg2():  # pylint: disable=too-many-statements
...
-                apt_install_cmd = 'apt-get install -y gcc libpq-dev python3-dev'.split()
-                if os.geteuid() != 0:  # pylint: disable=no-member
-                    apt_update_cmd.insert(0, 'sudo')
-                    apt_install_cmd.insert(0, 'sudo')
-                exit_code = subprocess.call(apt_update_cmd, True)
-                if exit_code == 0:
-                    logger.debug("Install dependencies with '%s'", " ".join(apt_install_cmd))
-                    subprocess.call(apt_install_cmd, True)
+            from azure.cli.core.util import in_cloud_console
+            if not in_cloud_console():
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/extension/operations.py:217]: The easiest code is to put this at the beginning of `_install_deps_for_psycopg2` at Line 185:

```py
if in_cloud_console():
    return
```

Comment (DaeunYim) [reply src/azure-cli-core/azure/cli/core/extension/operations.py]: That makes sense. Made the change accordingly. Thank you

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/extension/operations.py]: Please put these lines at L185.

**Review by jiasli — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/extension/operations.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/extension/operations.py]: I have made the change.

**Comment by jiasli** [comment]
Wait. https://github.com/Azure/azure-cli/issues/21535 reports the `sudo` error because **requirements aren't satisfied**. Doesn't https://github.com/Azure/CloudShell/pull/143 work?

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**